wrappers/python/nntile/model/generation/llm_samplers.py

```diff
@@ -0,0 +1,99 @@
+# @copyright (c) 2022-present Skolkovo Institute of Science and Technology
+#                              (Skoltech), Russia. All rights reserved.
+#                2023-present Artificial Intelligence Research Institute
+#                              (AIRI), Russia. All rights reserved.
+#
+# NNTile is software framework for fast training of big neural networks on
+# distributed-memory heterogeneous systems based on StarPU runtime system.
+#
+# @file wrappers/python/nntile/model/generation/llm_samplers.py
+#
+# @version 1.1.0
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+from scipy.special import softmax
+
+from nntile.model.generation.llm_params import GenerationMode
+
+
+def sample_topk(logits, k, temperature):
+    batch_extended = logits if len(logits.shape) > 1 else logits[:, None]
+    batch_indx = 0
+
+    sampled = np.empty((logits.shape[1],), dtype=np.int64)
+    for batch_indx in range(logits.shape[1]):
+        single_logits = batch_extended[:, batch_indx]
+        argsorted_logits = np.argsort(single_logits)
+        top_indices = argsorted_logits[-k:]
+        top_probas = softmax(single_logits[top_indices] / temperature, axis=0)
+        next_token = np.random.default_rng().choice(top_indices, p=top_probas)
+        sampled[batch_indx] = next_token
+    return sampled
+
+
+def sample_topp(logits, p_thr, temperature):
+    batch_extended = logits if len(logits.shape) > 1 else logits[:, None]
```

Comment:
the same comment, as with `sample_topk()` function

Reply:
done